R/parse-body.R

```diff
@@ -67,7 +77,7 @@ parser_picker <- function(content_type, first_byte, filename = NULL) {
     return(parsers$query)
```

Comment:
Do you think the parsers that are hard coded (`json`, `query`, `text`, `octet`) should always be available? (Pulled from `.globals$parsers, rather than `parsers` that's passed in.)

I'm leaning towards yes.  The only one that maybe not to be included is the `json` one. idk. Such as only do the first byte check on the json if the json parser is included. 🤷

Reply:
I've only added the one that were available in 0.4.6 (json and query). The other ones are opt-in.

I had them in initially but upon reading your comment, I thought this was the direction you wanted to go.  I lean towards not overriding what the user defined by using the .globals parsers.

Reply:
I like how you've set it up. It's great.  

Trying to figure out if there are some expected behavior that people **always** expect, regardless of what parsers are being used. 

Like `octet`, `query`, and `text` should **always** be available in the hard coded places in this function. I would even be fine if `text` was removed.

I'll update the defaults down below.

Reply:
Ok. I think I've caught up.  

You are correct.  The code above is clever in that it'll return early and possibly return NULL if the endpoint parser is missing.

The function that calls `parser_picker` will then check if a function was returned. Great!

(resolving)